browser/mapping.go

```diff
@@ -739,16 +739,16 @@ func mapBrowserContext(vu moduleVU, bc *common.BrowserContext) mapping { //nolin
 		"setOffline":         bc.SetOffline,
 		"storageState":       bc.StorageState,
 		"unroute":            bc.Unroute,
-		"waitForEvent": func(event string, optsOrPredicate goja.Value) *goja.Promise {
+		"waitForEvent": func(event string, optsOrPredicate goja.Value) (*goja.Promise, error) {
 			ctx := vu.Context()
-			return k6ext.Promise(ctx, func() (result any, reason error) {
-				parsedOpts := common.NewWaitForEventOptions(
-					bc.Timeout(),
-				)
-				if err := parsedOpts.Parse(ctx, optsOrPredicate); err != nil {
-					return nil, fmt.Errorf("parsing waitForEvent options: %w", err)
-				}
+			parsedOpts := common.NewWaitForEventOptions(
```

Comment:
Since we are using `popts` everywhere, should this be `popts` also?

Reply:
Yep, done in f8decf766086c81d3622a9a57b211e656b4257cd